modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/security/permission/resource/SegmentsEntryModelResourcePermissionWrapper.java

```diff
@@ -76,7 +78,10 @@ public Boolean contains(
 
 			if (actionId.equals(ActionKeys.UPDATE) &&
 				SegmentsEntryConstants.SOURCE_ASAH_FARO_BACKEND.equals(
-					segmentsEntry.getSource())) {
+					segmentsEntry.getSource()) &&
+				(!GetterUtil.getBoolean(
+					PropsUtil.get("feature.flag.LPS-172194")) ||
+				 (segmentsEntry.getCriteria() == null))) {
```

Comment:
Hey @cgoncas ,

This is returning an empty String when it is null as you can see in the impl :  

	public String getCriteria() {
		if (_criteria == null) {
			return "";
		}
		else {
			return _criteria;
		}
	}

So maybe we have to change it.

Thanks

Reply:
Totally true, changed!!